core/src/main/java/com/graphhopper/routing/profiles/TagParserFactory.java

```diff
@@ -0,0 +1,141 @@
+package com.graphhopper.routing.profiles;
+
+import com.graphhopper.reader.ReaderWay;
+import com.graphhopper.routing.util.AbstractFlagEncoder;
+import com.graphhopper.util.EdgeIteratorState;
+
+public class TagParserFactory {
```

Comment:
Isn't this factory a bit of an overkill? Wouldn't it be easier to put this into classes?